ouroboros-network/src/Ouroboros/Network/BlockFetch/Decision.hs

```diff
@@ -89,77 +89,144 @@ type FetchDecision result = Either FetchDecline result
 
 -- | All the various reasons we can decide not to fetch blocks from a peer.
 --
+-- It is worth highlighting which of these reasons result from competition
+-- among upstream peers.
+--
+-- * 'FetchDeclineInFlightOtherPeer': decline this peer because all the
+--   unfetched blocks of its candidate chain have already been requested
+--   fetched from other peers. This reason reflects the least-consequential
+--   competition among peers: the competition that determines merely which
+--   upstream peer to burden with the request (eg the one with the best
+--   'Ouroboros.Network.BlockFetch.DeltaQ.DeltaQ' metrics). The consequences
+--   are relatively minor because the unfetched blocks on this peer's candidate
+--   chain will be requested regardless; it's merely a question of "From who?".
+--   (One exception: if an adversarial peer wins this competition such that the
+--   blocks are only requested from them, then it may be possible that this
+--   decision determines whether the blocks are ever /received/. But that
+--   depends on details of timeouts, a longer competing chain being soon
+--   received within those timeouts, and so on.)
+--
+-- * 'FetchDeclineChainNotPlausible': decline this peer because the node has
+--   already fetched, validated, and selected a chain better than its candidate
+--   chain from other peers (or from the node's own block forge). Because the
+--   node's current selection is influenced by what blocks other peers have
+--   recently served (or it recently minted), this reason reflects that peers
+--   /indirectly/ compete by serving as long of a chain as possible and as
+--   promptly as possible. When the tips of the peers' selections are all
+--   within their respective forecast horizons (see
+--   'Ouroboros.Consensus.Ledger.SupportsProtocol.ledgerViewForecastAt'), then
+--   the length of their candidate chains will typically be the length of their
+--   selections, since ChainSync should severely outpace BlockFetch (both much
+--   smaller payloads and also much cheaper validation). But if one or more of
```

Comment:
By "validation", do you mean something like parsing the block CBOR in the context of BlockFetch?

Reply:
Thanks, I reworded this.